networkx/algorithms/covering.py

```diff
@@ -0,0 +1,100 @@
+"""
+********
+Covering
+********
+"""
+#    Copyright (C) 2015 by
+#    Nishant Nikhil <nishantiam@gmail.com>
+#    All rights reserved.
+#    BSD license.
+
+import networkx as nx
+from networkx.utils import not_implemented_for, arbitrary_element
+
+
+@not_implemented_for('directed', 'multigraph')
+def min_edge_cover(G):
+    """Returns a set of edges which constitute the minimum edge cover of the graph.
+    An edge cover of a graph is a set of edges such that every vertex of
+    the graph is incident to at least one edge of the set.
+    A minimum edge cover is an edge covering of smallest possible size.
+
+    `Edge Cover <https://en.wikipedia.org/wiki/Edge_cover>`_
+
+    A smallest edge cover can be found in polynomial time by finding
+    a maximum matching and extending it greedily so that all vertices
+    are covered.
+
+    Parameters
+    ----------
+    G : An undirected, unweighted graph.
+
+    Returns
+    -------
+    min_cover : set
+        The covering is returned as a set, ``min_cover``.
+        It contains all the edges of minimum edge cover
+        in form of tuples.
+
+    Notes
+    -----
+    Worst Case runtime of ``nx.max_weight_matching(G, maxcardinality=True)``
+    is `O(n^{3})`
+    And the greedy extension works in `O(n\log{}n)`
+    So worst case run time of the function is `O(n^{3})`.
+
+    Minimum edge cover for bipartite graph can also be found using the
+    function present in :mod:`networkx.algorithms.bipartite.covering`
+    """
+    if nx.number_of_isolates(G) > 0:
+        # ``min_cover`` does not exist as there is an isolated vertex
+        raise nx.NetworkXException(
+            "Graph has a vertex with no edge incident on it, "
+            "so no edge cover exists.")
+
+    maximum_matching = nx.max_weight_matching(G, maxcardinality=True)
+    # ``min_cover`` is superset of ``maximum_matching``
+    min_cover = set(maximum_matching.items())
+    # iterate for uncovered nodes
+    uncovered_nodes = set(G.nodes()) - set(v for u, v in min_cover)
+    for v in uncovered_nodes:
+        # Since `v` is uncovered, each edge incident to `v` will join it
+        # with a covered node (otherwise, if there were an edge joining
+        # uncovered nodes `u` and `v`, the maximum matching algorithm
+        # would have found it), so we can choose an arbitrary edge
+        # incident to `v`. (This applies only in a simple graph, not a
+        # multigraph.)
+        #
+        # Since the graph is guaranteed to be undirected, we need only
+        # add a single orientation of the edge.
+        u = arbitrary_element(G[v])
+        min_cover.add((u, v))
+        min_cover.add((v, u))
+    return min_cover
+
+
+def is_edge_covering(G, cover):
```

Comment:
In networkx/algorithms/tests/test_covering.py
```
def test_min_edge_cover_bipartite():
    G = nx.Graph()
    G.add_nodes_from([1, 2, 3, 4], bipartite=0)
    G.add_nodes_from(['a', 'b', 'c'], bipartite=1)
    G.add_edges_from([(1, 'a'), (1, 'b'), (2, 'b'),
                      (2, 'c'), (3, 'c'), (4, 'a')])
    min_cover = nx.min_edge_cover(G)
    assert_true(nx.is_edge_covering(G, min_cover))
    assert_equal(len(min_cover), 8)

def test_min_edge_cover_complete_graph():
    G = nx.complete_graph(10)
    min_cover = nx.min_edge_cover(G)
    assert_true(nx.is_edge_covering(G, min_cover))
    assert_equal(len(min_cover), 10)
```
As we are already doing `assert_true(nx.is_edge_covering(G, min_cover))`, Isn't it a sufficient test ?
PS: I realized the function name should be changed from `test_empty_graph()` to `test_min_edge_cover_empty_graph()`